### PR TITLE
Fix parsing request from backend logic in Chart component

### LINKED ISSUE #415: Temperature data did not displayed on the Chart
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in QA
    * Followed all applicable steps in the TROUBLESHOOTING guide: TODO
    * Checked for relevant discussions on Discord: <https://discord.gg/pPERUuv>
    * Checked that your issue isn't already filed: <https://github.com/COVID-19-electronic-health-system/Corona-tracker/issues>

### Description

Temperature data did not displayed on the Chart

Data being entered:
entered data

Empty Chart:
empty chart

### Steps to Reproduce

1) run ```yarn start```
2) click "Show me more"
3) See empty chart

**Expected behavior:**

We should see bars with temperature data for each day of entry.

**Actual behavior:**

Empty Chart

**Reproduces how often:**

100%

### PR DESCRIPTION
**Pull request by pavel-ilin**
fixes #415 

### Explain the **details** for making this change. What existing problem does the pull request solve?

- Add check for null values and empty strings
- Add all values to the final dataset

### Test plan (required)

yarn test

### Final Checklist

- [x] Have you bumped the version in `package.json`?
    - Second decimal for major change, third decimal for minor change. This can go past 10 i.e. 1.0.9 !=> 1.1.0, 1.0.9 => 1.0.10
    - [Read here for more info](https://semver.org/)
- [x] Have you added any new tests necessary?
- [x] Is your PR rebased off the most current master?
- [x] Have you squashed all commits, or if you will merge will you squash all commits?
- [x] Did you use yarn, not npm?
- [x] Did you use Material-UI wherever possible?
- [x] Did you format according to Prettier?
- [x] Did you run all of your most recent changes locally to make sure everything is working?

### CODE REVIEW

**Comment by AdhamAH** [comment]
as usual very neat!

File: client/src/components/Chart.jsx [hunk 1]
```diff
@@ -61,6 +61,10 @@ const Chart = ({ chartType }) => {
         data.map(dataPoint => {
           const date = new Date(dataPoint.date).toISOString().slice(0, 10);
           let temp = 0;
+          let numObservations = 1;
```
Comment (SomeMoosery) [inline client/src/components/Chart.jsx:64]: Why do we need to initialize `numObservations` here - this should be in Redux already no?

Comment (AdhamAH) [reply client/src/components/Chart.jsx:64]: I was not sure😅

Comment (pavel-ilin) [reply client/src/components/Chart.jsx]: Yep, we don't need it here.

File: client/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "corona-tracker",
-  "version": "0.2.28",
+  "version": "0.3.0",
```
Comment (SomeMoosery) [inline client/package.json:3]: Sorry I may have worded it poorly, we're at 0.3.0 now need to go to 0.3.1

Comment (pavel-ilin) [reply client/package.json]: Done

**Pull request merged**